src/js/Content/Features/Store/Common/FAddToCartNoRedirect.js

```diff
@@ -0,0 +1,54 @@
+import {SyncedStorage} from "../../../../modulesCore";
+import {Feature, RequestData} from "../../../modulesContent";
+
+export default class FAddToCartNoRedirect extends Feature {
+
+    checkPrerequisites() {
+        return SyncedStorage.get("addtocartnoredirect");
+    }
+
+    apply() {
+
+        function getFormEl(href) {
+
+            let formName;
+
+            const fnName = href.match(/javascript:([^(]+)/)[1];
+            switch (fnName) {
+                case "addToCart":
+                    formName = `add_to_cart_${href.match(/\d+/)}`;
+                    break;
+                case "addBundleToCart":
+                    formName = `add_bundle_to_cart_${href.match(/\d+/)}`;
+                    break;
+                case "GamePurchaseDropdownAddToCart":
+                    formName = `add_to_cart_${href.split("'")[1]}`;
+                    break;
+                case "addAllDlcToCart":
+                    formName = "add_all_dlc_to_cart";
+                    break;
+                default:
+                    return null;
+            }
+
+            return document.forms[formName];
+        }
+
+        document.addEventListener("click", async e => {
+
+            const node = e.target.closest("a[href^=javascript]");
+            if (!node) { return; }
+
+            const form = getFormEl(node.href);
+            if (!form) { return; }
+
+            e.preventDefault();
+
+            try {
+                await RequestData.post("https://store.steampowered.com/cart/", new FormData(form));
+            } catch (err) {} // ignore error
```

Comment:
TODO: Update cart icon here (maybe with animation), with an onhover event that shows a mini-cart. Just like this example on greenmangaming:
![image](https://user-images.githubusercontent.com/4411977/121067298-29e39000-c7cb-11eb-8878-15f5aa9620b4.png)